api/valor_api/backend/metrics/metric_utils.py

```diff
@@ -253,6 +256,126 @@ def create_metric_mappings(
     return ret
 
 
+def log_evaluation_duration(
+    db: Session,
+    evaluation: models.Evaluation,
+):
+    """
+    Store analytics regarding the evaluation's runtime in the metadata field of the evaluation table.
+
+    Parameters
+    ----------
+    db : Session
+        The database Session to query against.
+    evaluation : models.Evaluation
+        The evaluation to log to.
+    prediction_filter : schemas.Filter
+        The filter to be used to query predictions.
+    groundtruth_filter : schemas.Filter
+        The filter to be used to query groundtruths.
+    """
+
+    server_time = db.execute(func.now()).scalar().replace(tzinfo=None)  # type: ignore - guaranteed to return server time if psql is running
+    duration = (server_time - evaluation.created_at).total_seconds()
+
+    try:
+        metadata = dict(evaluation.meta) if evaluation.meta else {}
+        metadata.update({"duration": duration})
+        evaluation.meta = metadata
+        db.commit()
+    except IntegrityError as e:
+        db.rollback()
+        raise e
+
+
+def log_evaluation_item_counts(
+    db: Session,
+    evaluation: models.Evaluation,
+    prediction_filter: schemas.Filter,
+    groundtruth_filter: schemas.Filter,
+):
+    """
+    Store analytics regarding the number of elements processed by the evaluation in the metadata field of the evaluation table.
+
+    Parameters
+    ----------
+    db : Session
+        The database Session to query against.
+    evaluation : models.Evaluation
+        The evaluation to log to.
+    prediction_filter : schemas.Filter
+        The filter to be used to query predictions.
+    groundtruth_filter : schemas.Filter
+        The filter to be used to query groundtruths.
+    """
+    # get ground truth, prediction, annotation, and label counts
+    gt_subquery = (
+        Query(
+            models.Datum.id.label("datum_id"),
+            models.GroundTruth,
+        )
+        .filter(groundtruth_filter)
+        .groundtruths(as_subquery=False)
+        .alias()
+    )
+
+    gts = db.execute(
+        select(
+            gt_subquery.c.datum_id,
+            gt_subquery.c.annotation_id,
+            gt_subquery.c.label_id,
+        ).select_from(gt_subquery)
+    ).all()
```

Comment:
I thought i had left a comment about this but why cant we do something like this 

```
db.query(
    Query(
        func.count(distinct(models.Datum.id)),
        func.count(distinct(models.Annotation.id)),
        func.count(distinct(models.Label.id)),
    )
    .filter(groundtruth_filter)
    .groundtruths()
).all()
```

This would be more performant than counting in python